R/utilities.R

```diff
@@ -273,3 +187,58 @@ metaData2DataFrame <- function(metaData) {
 
   return(metaDF)
 }
+
+#' creates labels for percentiles
+#'
+#' convert a numeric percentile to a label
+#' e.g. `c(25,75)`   to `c("25th percentile", "75th percentile")`
+#'
+#' @param percentiles numeric input
+#' @param suffix common ending
+#'
+#' @return character vector with labels
+#' @export
+#'
+labelsForPercentile <- function(percentiles, suffix = " percentile") {
+  tmp <- lapply(
+    percentiles,
+    function(x) {
+      ending <- "th"
+      if (x %in% c(1, seq(21, 91, 10))) ending <- "st"
+      if (x %in% c(2, seq(22, 92, 10))) ending <- "nd"
+      if (x %in% c(3, seq(23, 93, 10))) ending <- "rd"
+      return(paste0(x, ending))
+    }
```

Comment:
Maybe we should use `scales::label_ordinal()` since `scales` is already a dependency:

``` r
labelsForPercentile <- function(percentiles, suffix = "percentile") {
  
  tmp <- scales::label_ordinal()(percentiles)
  
  paste(tmp, suffix)
}


labelsForPercentile(c(11, 22, 33, 50, 75))
#> [1] "11th percentile" "22nd percentile" "33rd percentile" "50th percentile"
#> [5] "75th percentile"
```

<sup>Created on 2023-12-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>